core/src/memory.rs

```diff
@@ -68,68 +63,83 @@ static_assertions::const_assert_eq!(WASM_PAGE_SIZE % GEAR_PAGE_SIZE, 0);
 static_assertions::const_assert_eq!(WASM_PAGE_SIZE % PAGE_STORAGE_GRANULARITY, 0);
 static_assertions::const_assert_eq!(PAGE_STORAGE_GRANULARITY % GEAR_PAGE_SIZE, 0);
 
+/// Alias for inner type of page buffer.
+pub type PageBufInner = LimitedVec<u8, (), GEAR_PAGE_SIZE>;
```

Comment:
Why do we need to do this? Is less performance effective .

Reply:
Please check comments on line 73 there.

What about performance: the vec is always preallocated before use so it shouldn't be impaired.

Reply:
ok see now